waiter/src/waiter/scheduler.clj

```diff
@@ -455,10 +457,11 @@
         (loop [service-id->health-check-context' {}
                healthy-service-ids #{}
                scheduler-messages []
-               [[service-id {:keys [active-instances failed-instances]}] & remaining] (seq (pc/map-keys :id service->service-instances))]
+               [[{service-id :id :as service} {:keys [active-instances failed-instances]}] & remaining] (seq service->service-instances)]
           (if service-id
             (let [request-instances-time (t/now)
                   active-instance-ids (->> active-instances (map :id) set)
+                  {:keys [instances task-count]} service
```

Comment:
Removed. See below.

Reply:
I added this back to support tracking app scaling. This monitoring loop now treats the initial instances created when a service is started uniformly with all other scale-up events.

`instances` here is used to populate the `:requested` task count field below. This is the count that we monitor to know when to start a *scheduling* instance timer. The `task-count` is the `:started` instance count below, which I'm currently only using for debugging. (I find it nice to have the started-count in the state so I can more easily observe the scale-up progress.)

Reply:
Regarding `active-instances`: taking that count should give the same number as we're getting from `task-count`—but since I'm only using that value for testing/debugging, I don't think it makes a big difference either way.